Tasks/Common/TlsHelper_/TlsHelper_.psm1

```diff
@@ -0,0 +1,53 @@
+function Add-Tls12InSession {
+    [CmdletBinding()]
+    param()
+
+    try {
+        if ([Net.ServicePointManager]::SecurityProtocol -notcontains 'Tls12') {
+            $securityProtocol=@()
+            $securityProtocol+=[Net.ServicePointManager]::SecurityProtocol
+            $securityProtocol+=[Net.SecurityProtocolType]3072
+            [Net.ServicePointManager]::SecurityProtocol=$securityProtocol
+
+            Write-Verbose 'Added TLS 1.2 in session.'
+        }
+        else {
+            Write-Verbose 'TLS 1.2 already present in session.'
+        }
+    }
+    catch {
+        Write-VstsTaskError 'Failed to add TLS 1.2 in session: $_.Exception.Message'
+    }
+}
+
+function Assert-TlsError {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)] $exception 
+    )
+
+    if ($exception -eq $null)
+    {
+        return
+    }
+
+    $isWebException = $true
+    $innerException = $exception
+    while ($innerException.GetType() -ne [System.Net.WebException])
```

Comment:
why don't we just use $exception.getbaseexception() here and avoid the loop altogether.

Reply:
getBaseException() gives the innermost exception only which may or may not be what we want.